### DISCUSSION TITLE
1 PR for npm, 1 for Maven, x for major updates

### DISCUSSION THREAD

**Discussion by nyg** [renovatebot/renovate#14315] [Request Help] [answered]
Hello,

I have a repository with one pom.xml and one package.json and I'm trying to set up the following rules:

1. one PR for all minor and patch updates for NPM dependencies,
2. one PR for all minor and patch updates for Maven dependencies,
3. x PRs for major version updates (independent of the package manager),
4. don't create more than 3 PRs.

What is important is that, if there a minor/patch updates, the first 2 PRs always be created.

I made the following config in a test repository (https://github.com/nyg/test-renovate) but it's not behaving as I expect:

<details>
<summary>renovate.json</summary>

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "packageRules": [
    {
      "matchManagers": ["npm"],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ]
    },
    {
      "matchManagers": ["maven"],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ]
    }
  ],
  "dependencyDashboard": true,
  "ignorePaths": [],
  "separateMultipleMajor": true,
  "prConcurrentLimit": 3
}
```
</details>

In the logs, I'm getting the warning `Each packageRule must contain at least one non-match* or non-exclude* field` for both package rules. But I don't understand what I should exclude 🤔.

Thanks for the help,
nyg

**Comment by rarkins** [comment, marked as answer]
Please read the documentation for `packageRules`

**Reply by nyg** [reply]
I re-read it but I still don't get it. Is it because match* within a packageRule are applied if either match? Which would mean the issue is that I have a `matchUpdateTypes` with the same value (minor/path) in two different rules?

**Reply by rarkins** [reply]
No, that's not it. The error is because you have included criteria in each package rule for how/what to match, but you have no additional config in each rule which says _what you want to do about it_ once there's a match (e.g. you might want to change the schedule, add a label, specify a reviewer, enable automerge, etc).

In your case you are wanting to group updates together, so add a `groupName` to each rule like in this example:

![image](https://user-images.githubusercontent.com/6311784/154860112-2fd5df04-642b-4667-a4d0-8678434b439c.png)

**Reply by nyg** [reply]
Ok, I feel a bit stupid now :). I had misunderstood that packageRules was not only for grouping packages… Thanks for your time, again.
